ideas/167-extended-automated-testing-for-continuous-delivery/README.md

```diff
@@ -0,0 +1,58 @@
+## Preamble
+
+    Idea: 167
+    Title: Functional automated testing as a part of Continuous Delivery process
+    Status: Draft
+    Created: 2018-04-12
+
+## Summary
+For automated testing to be a part of CD process we need to:
+- increase coverage
+- increase reliability
+- provide visible results on daily basis
+
+## Swarm Participants
+- Lead Contributor: @antdanchenko
+- Testing & Evaluation: @antdanchenko
+- Contributor: @yevh-berdnyk
+- Contributor: @lukaszfryc
+- PM:
+
+## Product Overview
+Extended set of automated tests which will run against each nightly build in order to provide results via TestRun in TestRail + short summary via Slack notification
+As result during nightly build testing a tester should execute only those tests which are not covered by automation in the same TestRun
+In case of failure in automated test a tester should add an appropriate status to the test (see `An option to review results should be added for each test via adding 4 statuses to TestRail` in `Product Description` section)
+Actual sanity test case which is running against each PR will keep running without changes
+
+
+### Product Description
+- Each automated test suite should be based on existing set of cases from TestRail for tracking percents of functional test coverage
+- Each automated test should be the reflection of a test case from TestRail
+- Results of each automated test from a build should be provided via TestRun in TestRail and linked to the appropriate case
```

Comment:
Will test cases in TestRail contain the test steps? If yes, what about the maintenance cost of having the test steps in both code and TestRail? Maybe TestRail test cases should only have titles and automated tests should log the steps that are being executed.

Reply:
Right now we don't support steps in most cases, so there is nothing to change. Expanding of test cases is not in scope of the idea